addon/-private/link-to-component.js

```diff
@@ -15,9 +15,16 @@ export default LinkComponent.extend({
     let owner = getOwner(this);
 
     if (owner.mountPoint) {
+      // Prepend engine mount point to targetRouteName
       let fullRouteName = owner.mountPoint + '.' + get(this, 'targetRouteName');
-
       set(this, 'targetRouteName', fullRouteName);
+
+      // Prepend engine mount point to current-when if set
+      let currentWhen = get(this, 'current-when');
+      if (currentWhen !== null) {
+        let fullCurrentWhen = owner.mountPoint + '.' + get(this, 'current-when');
```

Comment:
Not sure how I missed this...I explicitly created that variable so that `get()` didn't have to be executed multiple times. Will update now.